### PR TITLE
Predict song emotion - Part 1

### PR DESCRIPTION
**Pull request by krishrustagi**

### CODE REVIEW

File: emotunes-commons/src/main/java/com/emotunes/emotunes/dao/SongsDao.java [hunk 1]
```diff
@@ -15,4 +18,8 @@ public class SongsDao {
     public String addSong(SongMetadata songMetadata) {
         return songRepository.save(SongMapper.toSongEntity(songMetadata)).getId();
     }
+
+    public List<StoredSong> getAll() {
```
Comment (SarthakSavasil) [inline emotunes-commons/src/main/java/com/emotunes/emotunes/dao/SongsDao.java]: getAllSongs, findAllSongs krde

File: emotunes-admin/src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java [hunk 2]
```diff
@@ -189,4 +198,18 @@ private String getTitle(Tag tag) {
 
         return title;
     }
+
+    public void availAllSongsToUser(String userId, String trainingModelId) {
+        List<StoredSong> songList = songsDao.getAll();
+        songList.forEach(song -> {
+            Emotion songEmotion = null;
```
Comment (SarthakSavasil) [inline emotunes-admin/src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java]: 205 to 211 ek function bana sakta

File: emotunes-admin/src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java [hunk 3]
```diff
@@ -154,14 +158,19 @@ private String saveThumbnail(Tag tag) throws IOException {
...
+    private void availSongToAllUsers(String songId, String songUrl) {
         List<StoredUser> userList = userDao.findAll();
-        userList.forEach(
-                user -> {
-                    // todo: predict song by model id (userId);
-                    persistUserSongMapping(user.getId(), songId, Emotion.HAPPY);
-                }
-        );
+        for (StoredUser user : userList) {
+            Emotion songEmotion = null;
```
Comment (SarthakSavasil) [inline emotunes-admin/src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java]: yaha bhi hai same cheez cuz

File: emotunes-admin/src/main/java/com/emotunes/emotunes/service/impl/UserSongModelServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+        return Emotion.valueOf(response.trim());
+    }
+
+    private String runProcessBuilder(List<String> command) throws IOException {
+
+        StringBuilder response = new StringBuilder();
+        ProcessBuilder processBuilder = new ProcessBuilder(command);
+        Process process = processBuilder.start();
+        BufferedReader in = new BufferedReader(new InputStreamReader(process.getInputStream()));
```
Comment (SarthakSavasil) [inline emotunes-admin/src/main/java/com/emotunes/emotunes/service/impl/UserSongModelServiceImpl.java]: bufferedReader should also be closed using finally block

File: emotunes-admin/src/main/java/com/emotunes/emotunes/service/impl/UserSongModelServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        String response = runProcessBuilder(command);
+
+        return Emotion.valueOf(response.trim());
+    }
+
+    private String runProcessBuilder(List<String> command) throws IOException {
+
+        StringBuilder response = new StringBuilder();
+        ProcessBuilder processBuilder = new ProcessBuilder(command);
+        Process process = processBuilder.start();
```
Comment (SarthakSavasil) [inline emotunes-admin/src/main/java/com/emotunes/emotunes/service/impl/UserSongModelServiceImpl.java:39]: process builder aur process mai bhi dekh lena close ya destroy ya aisa kuch hota ho to

**Review by SarthakSavasil — CHANGES_REQUESTED** [review]

File: emotunes-admin/src/main/java/com/emotunes/emotunes/service/impl/UserSongModelServiceImpl.java (hunk 5 above)
Comment (krishrustagi) [reply emotunes-admin/src/main/java/com/emotunes/emotunes/service/impl/UserSongModelServiceImpl.java:39]: done

**Review by SarthakSavasil — APPROVED** [review]

**Pull request merged**